prow/pjutil/trigger.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pjutil
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowv1 "k8s.io/test-infra/prow/client/clientset/versioned/typed/prowjobs/v1"
+	prowconfig "k8s.io/test-infra/prow/config"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+)
+
+type prowjobResult struct {
+	Status pjapi.ProwJobState `json:"status"`
+	URL    string             `json:"prowjob_url"`
+}
+
+func resultForJob(pjclient prowv1.ProwJobInterface, selector string) (*prowjobResult, bool, error) {
+	w, err := pjclient.Watch(context.TODO(), metav1.ListOptions{FieldSelector: selector})
```

Comment:
Might as well use `context.Background()` since this explicitly not passing a context